examples/uefi/libvirt.tf

```diff
@@ -0,0 +1,36 @@
+provider "libvirt" {
+    uri = "qemu:///system"
+}
+
+resource "libvirt_volume" "ubuntu-cloud-uefi" {
+  name = "ubuntu-cloud-uefi"
+  source = "https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-uefi1.img"
+}
+
+resource "libvirt_volume" "volume" {
+  name = "vm${count.index}"
+  base_volume_id = "${libvirt_volume.ubuntu-cloud-uefi.id}"
+  count = 1
+}
+
+resource "libvirt_domain" "domain" {
+  count = 2
+  name = "ubuntu-cloud-${count.index}"
+  memory = "512"
+  vcpu = 1
+  # This is the file which will back the UEFI NVRAM content.
+  nvram = "/var/lib/libvirt/qemu/nvram/vm${count.index}_VARS.fd"
+  # This file needs to be provided by the user.
+  nvram_template = "/srv/provisioning/terraform/debian-stable-uefi_VARS.fd"
```

Comment:
Maybe we could surround these parameters in a common group. Something like:

```hcl
resource "libvirt_domain" "domain" {
  #...
  nvram {
    file = "/var/lib/libvirt/qemu/nvram/vm${count.index}_VARS.fd"
    template = "/srv/provisioning/terraform/debian-stable-uefi_VARS.fd"
  }
  #...
 }
```

Maybe the template could be provided as a real Terraform template (specially since it depends on the VM):

```hcl
data "template_file" "nvram_contents" {
  template = "${file("nvram/nvram.vars.tpl")}"

  vars {
    some_var = "1111"
  }
}

resource "libvirt_domain" "domain" {
  #...
  nvram {
    file = "/var/lib/libvirt/qemu/nvram/vm${count.index}_VARS.fd"
    template_contents = "${data.template_file.nvram_contents.rendered}"
  }
  #...
 }
```